src/runtime/thread_storage_scope.h

```diff
@@ -223,6 +232,7 @@ class ThreadAxisConfig {
         w.work_size[arg_index_map_[i]] = size;
       }
     }
+    w.dyn_shmem_size = static_cast<size_t>(x.values[base_ + arg_index_map_.size()].v_int64);
```

Comment:
We need to consider backward compatibility for kernels where dynamic shared memory is not used. 

Update https://github.com/apache/tvm/blob/main/src/runtime/meta_data.h#L103 to add a  `use_dyn_shared_memory`(default to false) field, update json reader to use DeclareOptionalField, to optionally read this field in. Only decode the value when the flag is true.

Reply:
done